### ISSUE TITLE
Membership-scoped functionality?

### ISSUE THREAD

**Issue by humblecoder** [santigarcor/laratrust#386]
- Laravel Version: 6.2
- Laratrust Version: 5.1?

### Description:
It seems that Laratrust "teams" are built around permissions more so than memberships.  Meaning, rather than having separate line items for "user/team", it conflates "role/user/team" and doesn't account for, perhaps, activity in the context of a specific team at a specific point in time.

Are there any plans for "membership-scoped" functionality rather than simply permission-per-team?

In this case it would require an ID/GUID for the actual role entry, and separate join for "membership" (rather than team_id directly).  This may require a bit more "coupling" of team functionality, but I think the ability to retain "membership-related" activity is important.

Hopefully the question makes sense.

**Comment by ChrisThompsonTLDR** [comment]
Your question is really hard to follow.

If a user has a role in a team, they have a "membership."

**Comment by humblecoder** [comment]
Consider, as an example, multiple memberships per person.  If an application I’m writing allows collaboration and multiple team memberships per person, I don’t really see a way forward with the current setup.  Perhaps that’s intentional, just curious.

**Comment by ChrisThompsonTLDR** [comment]
Laratrust allows a user to be on multiple teams and have different roles in each.

User | Team | Role
------------ | ------------- | -------------
Chris | Operations | Owner
Tom | Operations | Member
Chris | Design | Member
Tom | Design | Owner
Bob | Design | Member

You keep using the term "membership" but haven't given any example of what that is.

**Comment by humblecoder** [comment]
**TLDR;**  As it stands, the current `role_user` table uniques on multiple columns but doesn't appear to provide a single key by which to identify/use it.


Apologies.  I actually attempted to respond on mobile.  Anyway, a `membership` is basically `role_user` with an `id` column.  The purpose being, if I prefer to log in to a system and perform tasks **as** that user.  So, if I create a new project as `role_user`, I'd like to be able to track/audit/log that activity with that `membership_id` (or `role_user_id`).

Does that make any more sense?  🤔   Again, perhaps I simply missed something, or perhaps the current design was completely intentional.

**Comment by ChrisThompsonTLDR** [comment]
Attach an auto-increment field to the table?

**Comment by humblecoder** [comment]
Yeah, that's basically what it comes down to.  Just making sure there wasn't some clearer view I wasn't seeing.
